test/RadixSortTest.cc

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <cstdint>
+#include <limits>
+#include <gtest/gtest.h>
+
+#include "fbgemm/Utils.h"
+
+TEST(cpu_kernel_test, radix_sort_parallel_test) {
+  std::array<int, 8> keys = {1, 2, 4, 5, 4, 3, 2, 9};
+  std::array<int, 8> values = {0, 0, 0, 0, 1, 1, 1, 1};
+
+  int* sorted_keys;
+  int* sorted_values;
+
+  std::array<int, 8> keys_tmp;
+  std::array<int, 8> values_tmp;
+
+  std::tie(sorted_keys, sorted_values) = fbgemm::radix_sort_parallel(
+      keys.data(),
+      values.data(),
+      keys_tmp.data(),
+      values_tmp.data(),
+      keys.size(),
+      10);
+
+  std::array<int, 8> expect_keys_tmp = {1, 2, 2, 3, 4, 4, 5, 9};
+  std::array<int, 8> expect_values_tmp = {0, 0, 1, 1, 0, 1, 0, 1};
+  EXPECT_EQ(sorted_keys, keys_tmp.data());
+  EXPECT_EQ(sorted_values, values_tmp.data());
+  EXPECT_EQ(keys_tmp, expect_keys_tmp);
+  EXPECT_EQ(values_tmp, expect_values_tmp);
+}
+
+TEST(cpu_kernel_test, radix_sort_parallel_test_neg_vals) {
+  std::array<int64_t, 8> keys = {-4, -3, 0, 1, -2, -1, 3, 2};
+  std::array<int64_t, 8> values = {0, 0, 0, 0, 1, 1, 1, 1};
+
+  int64_t* sorted_keys;
+  int64_t* sorted_values;
+
+  std::array<int64_t, 8> keys_tmp;
+  std::array<int64_t, 8> values_tmp;
+
+  std::tie(sorted_keys, sorted_values) = fbgemm::radix_sort_parallel(
```

Comment:
Since we support C++17 now, you can use structured bindings instead.

```cpp
const auto [ sorted_keys, sorted_values ] = fbgemm::radix_sort_parallel(...)
```

Reply:
Good suggestion, thanks 😉